commands/volumes/common.go

```diff
@@ -0,0 +1,125 @@
+package volumecommands
+
+import (
+	"os"
+	"strings"
+
+	"github.com/gluster/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/servers/sunrpc"
+	"github.com/gluster/glusterd2/transaction"
+	"github.com/gluster/glusterd2/utils"
+	"github.com/gluster/glusterd2/volgen"
+	"github.com/gluster/glusterd2/volume"
+	"github.com/gluster/glusterd2/xlator"
+
+	"github.com/pborman/uuid"
+)
+
+func areOptionNamesValid(optsFromReq map[string]string) bool {
+
+	var xlOptFound bool
+	for o := range optsFromReq {
+
+		// assuming option to be of the form <domain>.<xlator-option>
+		// and <domain> will be the xlator type.
+		// Example: cluster/afr.eager-lock
+		// we know for certain that this isn't true
+
+		tmp := strings.Split(strings.TrimSpace(o), ".")
+		if len(tmp) != 2 {
+			return false
+		}
+		xlatorType := tmp[0]
+		xlatorOption := tmp[1]
+
+		options, ok := xlator.AllOptions[xlatorType]
+		if !ok {
+			return false
+		}
+
+		xlOptFound = false
+		for _, option := range options {
+			for _, key := range option.Key {
+				if xlatorOption == key {
+					xlOptFound = true
+				}
+			}
+		}
+		if !xlOptFound {
+			return false
+		}
+	}
+
+	return true
+}
+
+func generateVolfiles(c transaction.TxnCtx) error {
```

Comment:
This is moved from other file right? I am skipping review of this below functions in this file, is that fine?